app/models/activity.rb

```diff
@@ -0,0 +1,58 @@
+class Activity < ActiveRecord::Base
+  before_create { |activity| activity.event = generate_key(activity.object, activity.event) }
```

Comment:
I still think that you should store the `event`, and generate the `key` at run time.

Also you are mixing `key` and `event` here.